backend/src/main/java/com/htecgroup/skynest/repository/custom/impl/CassandraSaveWithTtlImpl.java

```diff
@@ -0,0 +1,25 @@
+package com.htecgroup.skynest.repository.custom.impl;
+
+import com.htecgroup.skynest.repository.custom.CassandraSaveWithTtl;
+import lombok.AllArgsConstructor;
+import org.springframework.data.cassandra.core.CassandraOperations;
+import org.springframework.data.cassandra.core.InsertOptions;
+import org.springframework.util.Assert;
+
+@AllArgsConstructor
+public class CassandraSaveWithTtlImpl<T> implements CassandraSaveWithTtl<T> {
+
+  private CassandraOperations operations;
+
+  @Override
+  public <S extends T> S save(S entity, int timeToLiveInSeconds) {
+
+    Assert.notNull(entity, "Entity must not be null");
```

Comment:
Instead of this, you can take a look at annotations such as _@NotNull_ and _@Min_ and apply that validation here

Reply:
I borrowed this code from the <code>save()</code> method from <code>SimpleCassandraRepository.java</code>. I'll switch to use annotations since they'll look cleaner.

Reply:
I've removed validation here as it's done in the service anyway.